File: src/main/java/net/hectus/neobb/shop/util/FilterState.java
package net.hectus.neobb.shop.util;

import net.hectus.neobb.util.Colors;
import net.kyori.adventure.text.format.TextColor;
import org.bukkit.Material;

public enum FilterState {
    YES(Material.LIME_STAINED_GLASS_PANE, Colors.POSITIVE),
    NO(Material.RED_STAINED_GLASS_PANE, Colors.NEGATIVE),
    UNSET(Material.GRAY_STAINED_GLASS_PANE, Colors.EXTRA);

    public final Material item;
    public final TextColor color;

    FilterState(Material item, TextColor color) {
        this.item = item;
        this.color = color;
    }
}

File: src/main/kotlin/net/hectus/neobb/shop/util/FilterState.kt
package net.hectus.neobb.shop.util

import net.hectus.neobb.util.Colors
import net.kyori.adventure.text.format.TextColor
import org.bukkit.Material

enum class FilterState(val item: Material, val color: TextColor) {
    YES(Material.LIME_DYE, Colors.POSITIVE),
    NO(Material.RED_DYE, Colors.NEGATIVE),
    UNSET(Material.WHITE_DYE, Colors.EXTRA)
}
